extensions/filter.c

```diff
@@ -483,6 +489,54 @@ filter_client_quit(void *data_)
 	/* No point in doing anything with ACT_KILL */
 }
 
+void
+filter_client_nick_change(void *data_)
+{
+	hook_data_nick_approval *data = data_;
+	struct Client *s = data->client;
+	if (IsOper(s)) {
+		return;
+	}
+
+	unsigned r = match_message("0", s, "NICK", NULL, data->nick);
+	if (r & ACT_DROP) {
+		data->approved = 0;
+	}
+	if (r & ACT_ALARM) {
+		sendto_realops_snomask(SNO_GENERAL, L_ALL | L_NETWIDE,
+			"FILTER: %s!%s@%s [%s]",
+			s->name, s->username, s->host, s->sockhost);
+	}
+	if (r & ACT_KILL) {
+		exit_client(NULL, s, s, FILTER_EXIT_MSG);
+	}
+}
+
+void
+filter_client_nick_set(void *data_)
+{
+	hook_data_nick_approval *data = data_;
+	struct Client *s = data->client;
+	struct sockaddr *addr = (void *)&s->localClient->ip;
+
+	if(find_conf_by_address(
```

Comment:
we can't make people immune from this by them being an oper, obviously, and this is the only alternative that came to mind

Reply:
or maybe we just treat it like a RESV. you have to connect first and then switch your nick